### PR TITLE
17.0 finetune pms anko

### PR DESCRIPTION
**Pull request by anko-odoo**

### CODE REVIEW

File: addons/payment_aps/models/payment_provider.py [hunk 1]
```diff
@@ -58,3 +60,11 @@ def _aps_calculate_signature(self, data, incoming=True):
...
         signing_string = ''.join([key, sign_data, key])
         return hashlib.sha256(signing_string.encode()).hexdigest()
+
+    def _get_default_payment_method_codes(self):
+        """ Override of `payment` to return the default payment method codes. """
+        default_codes = super()._get_default_payment_method_codes()
+        if self.code != 'aps':
+            return default_codes
+        return const.DEFAULT_PAYMENT_METHODS_CODES
+
```
Comment (AntoineVDV) [inline addons/payment_aps/models/payment_provider.py:70]: extra line
```suggestion
```

File: addons/payment_buckaroo/models/payment_provider.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 from odoo import fields, models
 
-from odoo.addons.payment_buckaroo.const import SUPPORTED_CURRENCIES
+from odoo.addons.payment_buckaroo.const import SUPPORTED_CURRENCIES, DEFAULT_PAYMENT_METHODS_CODES
```
Comment (AntoineVDV) [inline addons/payment_buckaroo/models/payment_provider.py:9]: Import only `const` and replace usages of `SUPPORTED_CURRENCIES` by `const.SUPPORTED_CURRENCIES`.

File: addons/payment_flutterwave/models/payment_provider.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 from odoo import _, api, fields, models
 from odoo.exceptions import ValidationError
 
-from odoo.addons.payment_flutterwave.const import SUPPORTED_CURRENCIES
+from odoo.addons.payment_flutterwave.const import SUPPORTED_CURRENCIES, DEFAULT_PAYMENT_METHODS_CODES
```
Comment (AntoineVDV) [inline addons/payment_flutterwave/models/payment_provider.py:12]: Same as for Buckaroo.

File: addons/payment_mercado_pago/const.py [hunk 4]
```diff
@@ -27,6 +27,17 @@
...
+# The codes of the payment methods to activate when Mercado Pago is activated.
+DEFAULT_PAYMENT_METHODS_CODES = [
+    # Primary payment methods.
+    'card',
+    # Brand payment methods.
+    'visa',
+    'mastercard',
+    'amex',
+    'discover',
+]
```
Comment (AntoineVDV) [inline addons/payment_mercado_pago/const.py:39]: Could you check if there isn't a card brand typically added by Mercado Pago? If so, it'd make sense to enable it.

File: addons/payment_mercado_pago/models/payment_provider.py [hunk 5]
```diff
@@ -9,7 +9,7 @@
 from odoo import _, fields, models
 from odoo.exceptions import ValidationError
 
-from odoo.addons.payment_mercado_pago.const import SUPPORTED_CURRENCIES
+from odoo.addons.payment_mercado_pago.const import SUPPORTED_CURRENCIES, DEFAULT_PAYMENT_METHODS_CODES
```
Comment (AntoineVDV) [inline addons/payment_mercado_pago/models/payment_provider.py:12]: Same as above.

File: addons/payment_mercado_pago/models/payment_provider.py [hunk 6]
```diff
@@ -78,3 +78,11 @@ def _mercado_pago_make_request(self, endpoint, payload=None, method='POST'):
...
             )
         return response.json()
+
+    def _get_default_payment_method_codes(self):
+        """ Override of `payment` to return the default payment method codes. """
+        default_codes = super()._get_default_payment_method_codes()
+        if self.code != 'mercado_pago':
+            return default_codes
+        return DEFAULT_PAYMENT_METHODS_CODES
+
```
Comment (AntoineVDV) [inline addons/payment_mercado_pago/models/payment_provider.py:88]: extra line
```suggestion
```

File: addons/payment_mollie/models/payment_provider.py [hunk 7]
```diff
@@ -9,7 +9,7 @@
 from odoo import _, fields, models, service
 from odoo.exceptions import ValidationError
 
-from odoo.addons.payment_mollie.const import SUPPORTED_CURRENCIES
+from odoo.addons.payment_mollie.const import SUPPORTED_CURRENCIES, DEFAULT_PAYMENT_METHODS_CODES
```
Comment (AntoineVDV) [inline addons/payment_mollie/models/payment_provider.py:12]: Same as above.

File: addons/payment_paypal/models/payment_provider.py [hunk 8]
```diff
@@ -4,7 +4,7 @@
 
 from odoo import _, fields, models
 
-from odoo.addons.payment_paypal.const import SUPPORTED_CURRENCIES
+from odoo.addons.payment_paypal.const import SUPPORTED_CURRENCIES, DEFAULT_PAYMENT_METHODS_CODES
```
Comment (AntoineVDV) [inline addons/payment_paypal/models/payment_provider.py:7]: Same as above.

File: addons/payment_payulatam/models/payment_provider.py [hunk 9]
```diff
@@ -5,7 +5,7 @@
 from odoo import fields, models
 from odoo.tools.float_utils import float_repr, float_split
 
-from odoo.addons.payment_payulatam.const import SUPPORTED_CURRENCIES
+from odoo.addons.payment_payulatam.const import SUPPORTED_CURRENCIES, DEFAULT_PAYMENT_METHODS_CODES
```
Comment (AntoineVDV) [inline addons/payment_payulatam/models/payment_provider.py:8]: Same as above.

File: addons/payment_razorpay/models/payment_provider.py [hunk 10]
```diff
@@ -11,7 +11,7 @@
 from odoo import _, fields, models
 from odoo.exceptions import ValidationError
 
-from odoo.addons.payment_razorpay.const import SUPPORTED_CURRENCIES
+from odoo.addons.payment_razorpay.const import SUPPORTED_CURRENCIES, DEFAULT_PAYMENT_METHODS_CODES
```
Comment (AntoineVDV) [inline addons/payment_razorpay/models/payment_provider.py:14]: Same as above.

File: addons/payment_sips/models/payment_provider.py [hunk 11]
```diff
@@ -5,7 +5,7 @@
 
 from odoo import fields, models
 
-from odoo.addons.payment_sips.const import SUPPORTED_CURRENCIES
+from odoo.addons.payment_sips.const import SUPPORTED_CURRENCIES, DEFAULT_PAYMENT_METHODS_CODES
```
Comment (AntoineVDV) [inline addons/payment_sips/models/payment_provider.py:8]: Same as above.

**Review by AntoineVDV — CHANGES_REQUESTED** [review]

**Review by AntoineVDV — APPROVED** [review]
@robodoo r+ rebase-ff
@robodoo delegate+

**Comment by anko-odoo** [comment]
@robodoo r+

**Comment by anko-odoo** [comment]
@robodoo r+

**Pull request closed without merging**